### ISSUE TITLE
Exec with CopyDir as PreBuildDependency

### ISSUE THREAD

**Issue by Leandros** [fastbuild/fastbuild#159]
Trying to specify a `CopyDir` node as one of `Exec`s `.PreBuildDependencies` fails with the following error message:

```
FASTBuild Error #1005 - Exec() - Unsupported node type in '.PreBuildDependencies'. (Node: 'res-models', Type: 'CopyDirNode')
Exec('res')
^
\--here
```

What's the reason behind this not being possible, and is there a sensible work-around? (I could think of a way, by using a self-supplied `cp` executable and therefore turning the `CopyDir` into a `Exec` node)

**Comment by ffulin** [comment]
There is no specific reason for this to be this way - it's just how the code has evolved over time. I can't think of any reason why this wouldn't be ok to enable.

Should be as simple as changing this in FunctionExec.cpp:
```C++
    if ( !GetNodeList( nodeGraph, funcStartIter, ".PreBuildDependencies", preBuildDependencies, false ) )
```
To:
```C++
    const bool allowCopyDirNodes = true;
    if ( !GetNodeList( nodeGraph, funcStartIter, ".PreBuildDependencies", preBuildDependencies, false, allowCopyDirNodes ) )
```

**Comment by Leandros** [comment]
Neat!
Will you include the change in the next version? That would be awesome!

**Comment by ffulin** [comment]
I'll try to get this into the next version. It'll be a little more complex than the above because it will require tests to be written.  Still, should not be too bad.  Let's keep this ticket open until this is an official release.

**Comment by ffulin** [comment]
I've implemented this change in the Perforce mainline and integrated it into the GitHub "dev" branch (https://github.com/fastbuild/fastbuild/commit/3025a038c1b6cc5feedce1c3177267ba91c57506). It will be in the next release (v0.93).  I'll leave this issue open until the new version is released.

In the end, I just ended up making this simple change as we described. A more holistic fix to this issue (i.e. the same problem with other combinations of nodes) is in progress as part of ongoing refactoring.
